src/com/amazon/ionelement/api/IonOperators.kt

```diff
@@ -0,0 +1,410 @@
+@IonOperators
+public infix fun FE.lte(other: FE): Boolean = this.doubleValue <= other.doubleValue
+@IonOperators
+public infix fun FE.gte(other: FE): Boolean = this.doubleValue >= other.doubleValue
+
+/* FloatElement with Double operators */
+@IonOperators
+public operator fun FE.plus(other: Double): FE = ionFloat(this.doubleValue + other, this.annotations, this.metas)
+@IonOperators
+public operator fun FE.minus(other: Double): FE = ionFloat(this.doubleValue - other, this.annotations, this.metas)
+@IonOperators
+public operator fun FE.times(other: Double): FE = ionFloat(this.doubleValue * other, this.annotations, this.metas)
+@IonOperators
+public operator fun FE.div(other: Double): FE = ionFloat(this.doubleValue / other, this.annotations, this.metas)
+@IonOperators
+public operator fun FE.rem(other: Double): FE = ionFloat(this.doubleValue % other, this.annotations, this.metas)
+@IonOperators
+public infix fun FE.eq(other: Double): Boolean = this.doubleValue == other
+@IonOperators
+public infix fun FE.lt(other: Double): Boolean = this.doubleValue < other
+@IonOperators
+public infix fun FE.gt(other: Double): Boolean = this.doubleValue > other
+@IonOperators
+public infix fun FE.lte(other: Double): Boolean = this.doubleValue <= other
+@IonOperators
+public infix fun FE.gte(other: Double): Boolean = this.doubleValue >= other
+
+/* Double with FloatElement operators */
+@IonOperators
+public operator fun Double.plus(other: FloatElement): FE = ionFloat(this + other.doubleValue, other.annotations, other.metas)
+@IonOperators
+public operator fun Double.minus(other: FloatElement): FE = ionFloat(this - other.doubleValue, other.annotations, other.metas)
+@IonOperators
+public operator fun Double.times(other: FloatElement): FE = ionFloat(this * other.doubleValue, other.annotations, other.metas)
+@IonOperators
+public operator fun Double.div(other: FloatElement): FE = ionFloat(this / other.doubleValue, other.annotations, other.metas)
+@IonOperators
+public operator fun Double.rem(other: FloatElement): FE = ionFloat(this % other.doubleValue, other.annotations, other.metas)
+@IonOperators
+public infix fun Double.eq(other: FloatElement): Boolean = this == other.doubleValue
+@IonOperators
+public infix fun Double.lt(other: FloatElement): Boolean = this < other.doubleValue
+@IonOperators
+public infix fun Double.gt(other: FloatElement): Boolean = this > other.doubleValue
+@IonOperators
+public infix fun Double.lte(other: FloatElement): Boolean = this <= other.doubleValue
+@IonOperators
+public infix fun Double.gte(other: FloatElement): Boolean = this >= other.doubleValue
+
+/* Text Element operators */
+@IonOperators
+public operator fun StringElement.plus(other: String): StringElement = ionString(this.textValue + other, this.annotations, this.metas)
+@IonOperators
+public operator fun SymbolElement.plus(other: String): SymbolElement = ionSymbol(this.textValue + other, this.annotations, this.metas)
```

Comment:
This allows you to do `SymbolElement + String` _and_ `SymbolElement + StringBuilder`.
```suggestion
public operator fun SymbolElement.plus(other: CharSequence): SymbolElement = ionString(this.textValue + other, this.annotations, this.metas)
```